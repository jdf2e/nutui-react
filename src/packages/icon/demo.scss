.nut-cell > .nutui-iconfont {
  margin-right: 10px;
}
.h5-ul {
  display: flex;
  flex-wrap: wrap;
  padding: 0;
  width: 100%;
}
.h5-ul .h5-li {
  flex: 0 0 25%;
  max-width: 25%;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
}
.h5-ul .h5-li .h5-span {
  height: 40px;
  font-size: 12px;
  text-align: center;
}
.h5-ul .h5-li .h5-span .nutui-iconfont {
  margin: 16px 0 16px;
}
