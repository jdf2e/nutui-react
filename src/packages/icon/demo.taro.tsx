import React, { useState } from 'react'
import Taro from '@tarojs/taro'
import '@nutui/icons-react-taro/dist/style_iconfont.css'
import { Cell, Toast } from '@nutui/nutui-react-taro'
import { useTranslate } from '@/sites/assets/locale/taro'
import Header from '@/sites/components/header'
import Demo1 from './demos/taro/demo1'
import Demo2 from './demos/taro/demo2'
import Demo3 from './demos/taro/demo3'
import Demo4 from './demos/taro/demo4'
import Demo5 from './demos/taro/demo5'
import Demo6 from './demos/taro/demo6'
import Demo7 from './demos/taro/demo7'
// import '@/packages/icon/demo.scss'

const IconDemo = () => {
  const [translated] = useTranslate({
    'zh-CN': {
      '84aa6bce': '基础用法',
      svg: 'SVG 按需使用',
      dab8a74f: '图片链接',
      '52c15454': '图标颜色',
      '7aeb5407': '图标大小',
      f2e6c6d6: '基础图标',
    },
    'zh-TW': {
      '84aa6bce': '基礎用法',
      svg: 'SVG 按需使用',
      dab8a74f: '圖片連結',
      '52c15454': '圖示顏色',
      '7aeb5407': '圖示大小',
      f2e6c6d6: '基礎圖示',
    },
    'en-US': {
      '84aa6bce': 'Basic Usage',
      svg: 'SVG import On Demand',
      dab8a74f: 'Image Link',
      '52c15454': 'IconFont Color',
      '7aeb5407': 'IconFont Size',
      f2e6c6d6: 'Base IconFont',
    },
  })

  const [state, setState] = useState({
    msg: '',
    type: 'text',
    cover: false,
    visible: false,
    duration: 2,
    closeOnOverlayClick: false,
    title: '',
    bottom: '',
    icon: '',
    center: true,
  })

  return (
    <>
      <Header />
      <div className={`demo ${Taro.getEnv() === 'WEB' ? 'web' : ''}`}>
        <Toast
          content={state.msg}
          visible={state.visible}
          type={state.type}
          duration={state.duration}
          icon={state.icon}
          closeOnOverlayClick={state.closeOnOverlayClick}
          onClose={() => {
            setState({
              ...state,
              visible: false,
            })
          }}
        />
        <h2>{translated.svg}</h2>
        <Cell>
          <Demo1 />
        </Cell>
        <h2>{translated['84aa6bce']}</h2>
        <Cell>
          <Demo2 />
        </Cell>
        <h2>{translated.dab8a74f}</h2>
        <Cell>
          <Demo3 />
        </Cell>
        <h2>{translated['52c15454']}</h2>
        <Cell>
          <Demo4 />
        </Cell>
        <h2>{translated['7aeb5407']}</h2>
        <Cell style={{ alignItems: 'center' }}>
          <Demo5 />
        </Cell>
        <Demo6 />
        <Demo7 />
      </div>
    </>
  )
}

export default IconDemo
