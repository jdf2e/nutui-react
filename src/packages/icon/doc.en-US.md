# Icon 图标

### 介绍

基于 IconFont 字体的图标集，可以通过 Icon 组件使用。

### 安装

``` javascript
import { Icon } from '@nutui/nutui-react';
```

## 代码演示

### 基础用法

`Icon` 的 `name` 属性支持传入图标名称或图片链接。

```tsx
<Icon name="dongdong"></Icon>
<Icon name="JD"></Icon>
<Icon size="40"  name="https://img11.360buyimg.com/imagetools/jfs/t1/137646/13/7132/1648/5f4c748bE43da8ddd/a3f06d51dcae7b60.png"></Icon>
```

### 图标颜色

`Icon` 的 `color` 属性用来设置图标的颜色。

```tsx
<Icon name="dongdong" color="#fa2c19"></Icon>
<Icon name="dongdong" color="#64b578"></Icon>
<Icon name="JD" color="#fa2c19"></Icon>
```

### 图标大小

`Icon` 的 `size` 属性用来设置图标的尺寸大小，默认单位为 `px`。

```tsx
<Icon name="dongdong"></Icon>
<Icon name="dongdong" size="24"></Icon>
<Icon name="dongdong" size="16"></Icon>
```

### 自定义图标

如果需要在现有 Icon 的基础上使用更多图标，可以引入第三方 iconfont 对应的字体文件和 CSS 文件，之后就可以在 Icon 组件中直接使用。

```css
/* 引入第三方或自定义的字体图标样式 */
@font-face {
  font-family: 'my-icon';
  src: url('./my-icon.ttf') format('truetype');
}

.my-icon {
  font-family: 'my-icon';
}

.my-icon-extra::before {
  content: '\e626';
}
```

```tsx
<!-- 通过 class-prefix 指定类名为 my-icon -->
<Icon class-prefix="my-icon" name="extra" />
```

## API

### Props

| 参数         | 说明                             | 类型             | 默认值           |
|--------------|----------------------------------|------------------|------------------|
| name         | 图标名称或图片链接               | String           | -                |
| color        | 图标颜色                         | String           | -                |
| size         | 图标大小，如 `20px` `2em` `2rem` | String or Number | -                |
| class-prefix | 类名前缀，用于使用自定义图标     | String           | `nutui-iconfont` |
| tag          | tsx 标签                        | String           | `i`              |

### Events

| 事件名 | 说明           | 回调参数     |
|--------|----------------|--------------|
| click  | 点击图标时触发 | event: Event |
