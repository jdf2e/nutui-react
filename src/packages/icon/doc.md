# Icon 图标

## 介绍

独立安装 @nutui/icons-react 图标组件包。提供两种使用方式（Svg 按需使用、IconFont 全量使用）。

## 安装

```bash
npm i --save @nutui/icons-react
```

## 代码演示

### 方式一： Svg 按需使用

按需加载组件使用方式，可选项见 @nutui/icons-react/dist/types/index.d.ts
:::demo

```tsx
import React from "react";
import { Add, Dongdong, UserAdd } from '@nutui/icons-react'

const App = () => {
  return <>
    <Add color="red" />
    <UserAdd />
    <Dongdong />
  </>
}

```

:::

## Icon

### Props

| 属性 | 说明 | 类型 | 默认值 |
| --- | --- | --- | --- |
| name | 图标名称或图片链接 | `string` | `-` |
| color | 图标颜色 | `string` | `-` |
| width | 图标大小，如 `20px` `2em` `2rem` | `string` \| `object` | `-` |
| height | 图标大小，如 `20px` `2em` `2rem` | `string` \| `object` | `-` |
| onClick | 点击图标时触发 | `event: Event` | `-` |

## 方式二： IconFont 全量使用

```tsx
import { IconFont } from '@nutui/icons-react'
```

### 基础用法

`Icon` 的 `name` 属性支持传入图标名称或图片链接。

:::demo

```tsx
import React from "react";
import { IconFont } from '@nutui/icons-react'

const App = () => {
  return <>
    <IconFont name="dongdong" />
    <IconFont name="add" />
    <IconFont name="minus" />
  </>
}
export default App;

```

:::

### 图片链接

`Icon` 的 `name` 属性支持传入图标名称或图片链接。

:::demo

```tsx
import React from "react";
import { IconFont } from '@nutui/icons-react'

const App = () => {
  return <>
    <IconFont
      size="40"
      name="https://img11.360buyimg.com/imagetools/jfs/t1/137646/13/7132/1648/5f4c748bE43da8ddd/a3f06d51dcae7b60.png"
    />
  </>
}
export default App;

```

:::

### 图标颜色

`Icon` 的 `color` 属性用来设置图标的颜色。

:::demo

```tsx
import React from "react";
import { IconFont } from '@nutui/icons-react'

const App = () => {
  return <>
    <IconFont name="dongdong" color="#fa2c19" />
    <IconFont name="dongdong" color="#64b578" />
    <IconFont name="dongdong" color="#ffd700" />
  </>
}

export default App;
```

:::

### 图标大小

`Icon` 的 `size` 属性用来设置图标的尺寸大小，默认单位为 `px`。

:::demo

```tsx
import React from "react";
import { IconFont } from '@nutui/icons-react'

const App = () => {
  return <>
    <IconFont name="dongdong" size="16" />
    <IconFont name="dongdong" size="20" />
    <IconFont name="dongdong" size="24" />
  </>
}

export default App;
```

:::

### 自定义图标

如果需要在现有 Icon 的基础上使用更多图标，可以引入第三方 iconfont 对应的字体文件和 CSS 文件，之后就可以在 Icon 组件中直接使用。

> 方案一 引入 [iconfont](https://www.iconfont.cn/) 推荐此方案

第一步：首先在 [iconfont](https://www.iconfont.cn/) 生成你自定义的Icon文件下载存放至本地项目 [详细使用说明](https://www.iconfont.cn/help/detail?spm=a313x.7781069.1998910419.d8d11a391&helptype=code)

```bash
/assets/font/demo.css
/assets/font/demo_index.html
/assets/font/iconfont.css
/assets/font/iconfont.js
/assets/font/iconfont.json
/assets/font/iconfont.ttf
/assets/font/iconfont.woff
/assets/font/iconfont.woff2
```

第二步：项目入口文件 main.js 引用 `iconfont.css`

```tsx
import './assets/font/iconfont.css';
```

第三步:

```tsx
// fontClassName 指定类名为默认 iconfont
// classPrefix 指定默认 icon
// name 值根据 iconfont.css 中值对应填写 
import React from 'react'
import { IconFont } from '@nutui/icons-react'

const App = () => {
  return <IconFont fontClassName="iconfont" classPrefix='icon' name="close"/>
}
```

> 方案二 第三方自定义字体库

```css
/* 引入第三方或自定义的字体图标样式 */
@font-face {
  font-family: 'my-icon';
  src: url('./my-icon.ttf') format('truetype');
}

.my-icon {
  font-family: 'my-icon';
}

.icon-extra::before {
  content: '\e626';
}
```

```tsx
import React from "react";
import { IconFont } from '@nutui/icons-react'

const App = () => {
  return <>
    <IconFont fontClassName="my-icon" classPrefix="icon" name="extra" />
  </>
}

export default App;
```

## IconFont

### Props

| 属性 | 说明 | 类型 | 默认值 |
| --- | --- | --- | --- |
| name | 图标名称或图片链接 | `string` | `-` |
| color | 图标颜色 | `string` | `-` |
| size | 图标大小，如 `20px` `2em` `2rem` | `string` \| `number` | `-` |
| classPrefix | 类名前缀，用于使用自定义图标 | `string` | `nut-iconfont` |
| fontClassName | 自定义 icon 字体基础类名 | `string` | `nutui-iconfont` |
| tag | tsx 标签 | `string` | `i` |
| onClick | 点击图标时触发 | `event: Event` | `-` |

## 主题定制

### 样式变量

组件提供了下列 CSS 变量，可用于自定义样式，使用方法请参考 [ConfigProvider 组件](#/zh-CN/component/configprovider)。

| 名称 | 说明 | 默认值 |
| --- | --- | --- |
| \--nutui-icon-height | iconfont 容器的高度 | `16px` |
| \--nutui-icon-width | iconfont 容器的宽度 | `16px` |
| \--nutui-icon-line-height | iconfont 的行高 | `16px` |
