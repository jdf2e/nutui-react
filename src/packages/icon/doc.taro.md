# Icon 图标

### 介绍

基于 IconFont 字体的图标集，可以通过 Icon 组件使用。

### 安装

``` javascript
import { Icon } from '@nutui/nutui-react-taro';
```

## 代码演示

### 基础用法

`Icon` 的 `name` 属性支持传入图标名称或图片链接。

:::demo
```tsx
import React from "react";
import { Icon } from '@nutui/nutui-react-taro';

const App = () => {
  return <>
    <Icon name="dongdong" />
    <Icon name="JD"/>
    <Icon size="40"  name="https://img11.360buyimg.com/imagetools/jfs/t1/137646/13/7132/1648/5f4c748bE43da8ddd/a3f06d51dcae7b60.png"/>
  </>
}
export default App;

```
:::

### 图标颜色

`Icon` 的 `color` 属性用来设置图标的颜色。

:::demo
```tsx
import React from "react";
import { Icon } from '@nutui/nutui-react-taro';

const App = () => {
  return <>
    <Icon name="dongdong" color="#fa2c19" />
    <Icon name="dongdong" color="#64b578" />
    <Icon name="JD" color="#fa2c19" />
  </>
}

export default App;
```
:::

### 图标大小

`Icon` 的 `size` 属性用来设置图标的尺寸大小，默认单位为 `px`。

:::demo
```tsx
import React from "react";
import { Icon } from '@nutui/nutui-react-taro';

const App = () => {
  return <>
    <Icon name="dongdong" />
    <Icon name="dongdong" size="24" />
    <Icon name="dongdong" size="16" />
  </>
}

export default App;
```
:::
### 自定义图标

如果需要在现有 Icon 的基础上使用更多图标，可以引入第三方 iconfont 对应的字体文件和 CSS 文件，之后就可以在 Icon 组件中直接使用。

> 方案一 引入 [iconfont](https://www.iconfont.cn/)   推荐此方案

第一步：首先在 [iconfont](https://www.iconfont.cn/) 生成你自定义的Icon文件下载存放至本地项目  [详细使用说明](https://www.iconfont.cn/help/detail?spm=a313x.7781069.1998910419.d8d11a391&helptype=code)

``` bash
/assets/font/demo.css
/assets/font/demo_index.html
/assets/font/iconfont.css
/assets/font/iconfont.js
/assets/font/iconfont.json
/assets/font/iconfont.ttf
/assets/font/iconfont.woff
/assets/font/iconfont.woff2
```

第二步：项目入口文件 main.js 引用 `iconfont.css`


``` javascript
import './assets/font/iconfont.css';
```

第三步:

```tsx
// fontClassName 指定类名为默认 iconfont
// classPrefix 指定默认 icon
// name 值根据 iconfont.css 中值对应填写 
import React from 'react'
import Icon from '@nutui/nutui-react-taro'

const App = () => {
  return <Icon fontClassName="iconfont" classPrefix='icon' name="close"/>
}
```

> 方案二 第三方自定义字体库

```css
/* 引入第三方或自定义的字体图标样式 */
@font-face {
  font-family: 'my-icon';
  src: url('./my-icon.ttf') format('truetype');
}

.my-icon {
  font-family: 'my-icon';
}

.icon-extra::before {
  content: '\e626';
}
```

```tsx
import React from "react";
import { Icon } from '@nutui/nutui-react-taro';

const App = () => {
  return <>
    <Icon fontClassName="my-icon" classPrefix="icon" name="extra" />
  </>
}

export default App;
```

## API

### Props

| 参数          | 说明                             | 类型             | 默认值           |
|-------------|----------------------------------|------------------|------------------|
| name        | 图标名称或图片链接               | String           | -                |
| color       | 图标颜色                         | String           | -                |
| size        | 图标大小，如 `20px` `2em` `2rem` | String or Number | -                |
| classPrefix | 类名前缀，用于使用自定义图标     | String           | `nut-iconfont` |
| fontClassName           | 自定义 icon 字体基础类名                        | String           | `nutui-iconfonti`              |
| tag         | tsx 标签                        | String           | `i`              |

### Events

| 事件名           | 说明           | 回调参数     |
|---------------|----------------|--------------|
| onClick`v1.3.8` | 点击图标时触发 | event: Event |


## 主题定制

### 样式变量

组件提供了下列 CSS 变量，可用于自定义样式，使用方法请参考 [ConfigProvider 组件](#/zh-CN/component/configprovider)。

| 名称 | 默认值 |
| --- | --- |
| --nutui-icon-height | ` 20px` |
| --nutui-icon-width | ` 20px` |
| --nutui-icon-line-height | ` 20px` |
