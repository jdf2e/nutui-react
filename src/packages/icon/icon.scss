@import '../../styles/font/iconfont.css';
@import '../../styles/mixins/index.scss';
@import '../../styles/animation/index.scss';

.nut-icon {
  display: inline-block;
  position: relative;
  width: $icon-width;
  height: $icon-height;
  line-height: $icon-line-height;
  text-align: right;

  &::before {
    position: absolute;
    top: 50%;
    left: 50%;
    transform: translate(-50%, -50%);
  }

  &__img {
    width: $icon-width;
    height: $icon-height;
    object-fit: contain;
  }
  &-loading,
  &-loading1 {
    display: inline-block;
    animation: rotation 1s infinite linear;
  }

  --animate-duration: 1s;
  --animate-delay: 0s;

  &-am-infinite {
    -webkit-animation-iteration-count: infinite;
    -webkit-animation-direction: alternate;
    animation-iteration-count: infinite;
    animation-direction: alternate;
  }
  &-am-jump {
    -webkit-animation-name: nutJumpOne;
    -webkit-animation-duration: var(--animate-duration);
    -webkit-animation-timing-function: ease;
    -webkit-animation-delay: var(--animate-delay);

    animation-name: nutJumpOne;
    animation-duration: var(--animate-duration);
    animation-timing-function: ease;
    animation-delay: var(--animate-delay);

    &.nut-icon-am-infinite {
      -webkit-animation-name: nutJump;
      animation-name: nutJump;
    }
  }

  &-am-rotate {
    -webkit-animation-name: rotation;
    -webkit-animation-duration: var(--animate-duration);
    -webkit-animation-timing-function: linear;
    -webkit-animation-delay: var(--animate-delay);

    animation-name: rotation;
    animation-duration: var(--animate-duration);
    animation-timing-function: linear;
    animation-delay: var(--animate-delay);

    &.nut-icon-am-infinite {
      -webkit-animation-direction: normal;
      animation-direction: normal;
    }
  }
  &-am-blink {
    -webkit-animation-name: nutBlink;
    -webkit-animation-duration: var(--animate-duration);
    -webkit-animation-timing-function: ease-in-out;
    -webkit-animation-delay: var(--animate-delay);

    animation-name: nutBlink;
    animation-duration: var(--animate-duration);
    animation-timing-function: linear;
    animation-delay: var(--animate-delay);
  }
  &-am-breathe {
    -webkit-animation-name: nutBreathe;
    -webkit-animation-duration: var(--animate-duration);
    -webkit-animation-timing-function: ease-in-out;
    -webkit-animation-delay: var(--animate-delay);

    animation-name: nutBreathe;
    animation-duration: var(--animate-duration);
    animation-timing-function: ease-in-out;
    animation-delay: var(--animate-delay);
  }
  &-am-flash {
    -webkit-animation-name: nutFlash;
    -webkit-animation-duration: var(--animate-duration);
    -webkit-animation-timing-function: ease-in-out;
    -webkit-animation-delay: var(--animate-delay);

    animation-name: nutFlash;
    animation-duration: var(--animate-duration);
    animation-timing-function: ease-in-out;
    animation-delay: var(--animate-delay);
  }
  &-am-bounce {
    -webkit-animation-name: nutBounce;
    -webkit-animation-duration: var(--animate-duration);
    -webkit-animation-timing-function: ease-in-out;
    -webkit-animation-delay: var(--animate-delay);
    animation-name: nutBounce;
    animation-duration: var(--animate-duration);
    animation-timing-function: ease-in-out;
    animation-delay: var(--animate-delay);

    &.nut-icon-am-infinite {
      -webkit-animation-direction: normal;
      animation-direction: normal;
    }
  }

  &-am-shake {
    -webkit-animation-name: nutShake;
    -webkit-animation-duration: var(--animate-duration);
    -webkit-animation-timing-function: ease-in-out;
    -webkit-animation-delay: var(--animate-delay);

    animation-name: nutShake;
    animation-duration: var(--animate-duration);
    animation-timing-function: ease-in-out;
    animation-delay: var(--animate-delay);
  }
}
