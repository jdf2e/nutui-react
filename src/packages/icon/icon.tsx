import React, { ReactElement, ReactHTML } from 'react'
import bem from '@/utils/bem'

export interface IconProps {
  name: string
  size: string | number
  classPrefix: string
  color: string
  tag: keyof ReactHTML
  onClick: (e: MouseEvent) => void
  fontClassName: string
  className: string
  style: React.CSSProperties
  children: React.ReactNode
}

const defaultProps = {
  name: '',
  size: '',
  classPrefix: 'nut-icon',
  fontClassName: 'nutui-iconfont',
  color: '',
  tag: 'i',
  onClick: (e: MouseEvent) => {},
  className: '',
} as IconProps

function pxCheck(value: string | number): string {
  if (value === '') {
    return value
  }
  if (Number.isNaN(Number(value))) {
    return String(value)
  }
  return `${value}px`
}

export function Icon<T>(props: Partial<IconProps> & T): ReactElement {
  const {
    name,
    size,
    classPrefix,
    color,
    tag,
    children,
    className,
    fontClassName,
    style,
    onClick,
    ...rest
  } = {
    ...defaultProps,
    ...props,
  }
  const isImage = name ? name.indexOf('/') !== -1 : false
  const type = isImage ? 'img' : tag
  const b = bem('icon')

  const handleClick = (e: MouseEvent) => {
    if (onClick) {
      onClick(e)
    }
  }
  const hasSrc = () => {
    if (isImage) return { src: name }
    return {}
  }

  const styles: any = {}
  const checkedSize: any = pxCheck(size)
  if (checkedSize) {
    styles.width = styles.height = styles.fontSize = checkedSize
  }

  return React.createElement<any>(
    type,
    {
      className: isImage
        ? `${b('img')} ${className || ''} `
        : `${fontClassName} ${b(null)} ${classPrefix}-${name} ${
            className || ''
          }`,
      style: {
        color,
        ...styles,
        ...style,
      },
      ...rest,
      onClick: handleClick,
      ...hasSrc(),
    },
    children
  )
}

Icon.defaultProps = defaultProps
Icon.displayName = 'NutIcon'
