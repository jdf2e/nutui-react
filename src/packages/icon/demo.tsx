import React from 'react'
import { useTranslate } from '../../sites/assets/locale'
import { Icon } from './icon'
import Cell from '../cell'
import CellGroup from '../cellgroup'
import Toast from '../toast'
import icons from '@/styles/font/config.json'

interface T {
  '84aa6bce': string
  dab8a74f: string
  '52c15454': string
  '7aeb5407': string
  f2e6c6d6: string
}
const generateCopyText = (name: string) => {
  return `<Icon name="${name}"></Icon>`
}
const generateAMCopyText = (icon: any) => {
  return `
  <Icon name="${icon.name}"
    className="${`nut-icon-${icon['animation-name']}  nut-icon-${icon['animation-time']}`}"/>`
}
const copyTag = (text: string) => {
  const input = document.createElement('input')
  document.body.appendChild(input)
  input.setAttribute('value', text)
  input.select()
  if (document.execCommand('copy')) {
    document.execCommand('copy')
    Toast.text(`Copy: ${text}`)
  }
  document.body.removeChild(input)
}

const style = `
.nut-cell > .nutui-iconfont {
  margin-right: 10px;
}
ul {
  display: flex;
  flex-wrap: wrap;
  padding: 0;
  width: 100%;
}
ul li {
    flex: 0 0 25%;
    max-width: 25%;
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
}
ul li  span {
  height: 40px;
  font-size: 12px;
  text-align: center;
}
ul li span .nutui-iconfont {
  margin: 16px 0 16px;
}
`

const IconDemo = () => {
  const [translated] = useTranslate<T>({
    'zh-CN': {
      '84aa6bce': '基础用法',
      dab8a74f: '图片链接',
      '52c15454': '图标颜色',
      '7aeb5407': '图标大小',
      f2e6c6d6: '基础图标',
    },
    'zh-TW': {
      '84aa6bce': '基礎用法',
      dab8a74f: '圖片連結',
      '52c15454': '圖示顏色',
      '7aeb5407': '圖示大小',
      f2e6c6d6: '基礎圖示',
    },
    'en-US': {
      '84aa6bce': 'Basic Usage',
      dab8a74f: 'Image link',
      '52c15454': 'Icon color',
      '7aeb5407': 'Icon size',
      f2e6c6d6: 'Base Icon',
    },
  })

  return (
    <>
      <style>{style}</style>
      <div className="demo">
        <h2>{translated['84aa6bce']}</h2>
        <Cell>
          <Icon name="dongdong" />
          <Icon name="JD" />
        </Cell>
        <h2>{translated.dab8a74f}</h2>
        <Cell>
          <Icon
            size="40"
            name="https://img11.360buyimg.com/imagetools/jfs/t1/137646/13/7132/1648/5f4c748bE43da8ddd/a3f06d51dcae7b60.png"
          />
        </Cell>
        <h2>{translated['52c15454']}</h2>
        <Cell>
          <Icon name="dongdong" color="#fa2c19" />
          <Icon name="dongdong" color="#64b578" />
          <Icon name="JD" color="#fa2c19" />
        </Cell>
        <h2>{translated['7aeb5407']}</h2>
        <Cell>
          <Icon name="dongdong" />
          <Icon name="dongdong" size="24" />
          <Icon name="dongdong" size="16" />
        </Cell>
        {icons.data.map((item, index) => {
          return (
            <CellGroup key={index} title={item.name}>
              <Cell>
                <ul>
                  {item.icons.map((icon) => {
                    return (
                      <li
                        key={icon}
                        onClick={() => copyTag(generateCopyText(icon))}
                      >
                        <Icon name={icon} />
                        <span>{icon}</span>
                      </li>
                    )
                  })}
                </ul>
              </Cell>
            </CellGroup>
          )
        })}
        {icons.style.map((item, index) => {
          return (
            <CellGroup key={index} title={item.name}>
              <Cell>
                <ul>
                  {item.icons.map((icon) => {
                    return (
                      <li
                        key={icon.name}
                        onClick={() => copyTag(generateAMCopyText(icon))}
                      >
                        <Icon
                          name={icon.name}
                          className={`nut-icon-${icon['animation-name']}  nut-icon-${icon['animation-time']}`}
                        />
                        <span>{icon['animation-name']}</span>
                      </li>
                    )
                  })}
                </ul>
              </Cell>
            </CellGroup>
          )
        })}
      </div>
    </>
  )
}

export default IconDemo
