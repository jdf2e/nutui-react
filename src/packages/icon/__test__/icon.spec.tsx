import * as React from 'react'
import { fireEvent, render } from '@testing-library/react'
import '@testing-library/jest-dom'
import { Icon } from '../icon'

test('should icon snapshot match', () => {
  const { asFragment } = render(
    <Icon className="test" name="JD" color="#fa2c19" size="24" />
  )
  expect(asFragment()).toMatchSnapshot()
})

test('should color #fa2c19', async () => {
  const { container } = render(
    <Icon className="test" name="JD" color="#fa2c19" size="24" />
  )
  expect(container.querySelector('i')?.style.color).toBeTruthy()
})

test('should classname test', async () => {
  const { container } = render(
    <Icon className="test" name="JD" color="#fa2c19" size="24" />
  )
  expect(container.querySelector('.test')).toBeInTheDocument()
})

test('should onclick called', () => {
  const handleClick = jest.fn(() => {})
  const { getByTestId } = render(
    <Icon
      data-testid="test"
      className="test"
      name="JD"
      color="#fa2c19"
      size="24"
      click={handleClick}
    />
  )
  fireEvent.click(getByTestId('test'))
  expect(handleClick).toBeCalled()
})
