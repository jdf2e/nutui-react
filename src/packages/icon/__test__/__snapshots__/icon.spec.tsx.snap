// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should icon snapshot match 1`] = `
<DocumentFragment>
  <i
    class="nutui-iconfont nut-icon nut-icon-JD test"
    style="color: rgb(250, 44, 25); font-size: 24px; width: 24px; height: 24px;"
  />
</DocumentFragment>
`;
